### DISCUSSION TITLE
Unauthorized for RTSP Source

### DISCUSSION THREAD

**Discussion by pedromoraesh** [bluenviron/mediamtx#658] [Questions]
Hi guys,

I Did try to add a stream using HTTP API by running:

`curl -X POST -d '{"source": "rtsp://myuser:mypass@192.168.1.2:50554/live/mpeg4", "readUser": "myuser", "readPass": "mypass"}' http://localhost:9997/v1/config/paths/add/mypath`

and

`curl -X POST -d '{"source": "rtsp://myuser:mypass@192.168.1.2:50554/live/mpeg4"}' http://localhost:9997/v1/config/paths/add/mypath`

I also tried another path `/cam/realmonitor?channel=1&subtype=0` instead of `/live/mpeg4`.

It adds, but it keeps showing 

![image](https://user-images.githubusercontent.com/16081608/138357772-462c66d1-b3d1-4c74-9ca1-81490adabd9c.png)

I can open the stream by using Python3's OpenCV implementation. I think the stream uses h264 to compress the video according to the manual.

**Comment by aler9** [comment]
Hello, the error message means that you're using wrong credentials to access your camera. Please try with the correct set of username and password, and if it still doesn't work open an issue and provide a network dump.

**Reply by pedromoraesh** [reply]
> Hello, the error message means that you're using the wrong credentials to access your camera. Please try with the correct set of username and password, and if it still doesn't work open an issue and provide a network dump.

I opened the stream using Python's OpenCV, but with the server, the error keeps showing. I will try a few more before opening an issue.

[a comment by an automated account is not shown]
